Shared/Tracking/Trackers/shikimori/ShikimoriTracker.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  ShikimoriTracker.swift
+//  Aidoku
+//
+//  Created by Vova Lapskiy on 02.11.2024.
+//
+
+import Foundation
+
+#if os(OSX)
+    import AppKit
+#else
+    import UIKit
+#endif
+
+class ShikimoriTracker: OAuthTracker {
+    var api = ShikimoriApi()
+    let callbackHost = "shikimori-auth"
+    var oauthClient: OAuthClient { api.oauth }
+    lazy var authenticationUrl: String = api.getAuthenticationUrl() ?? ""
+
+    var id: String = "shikimori"
+    var name: String = "Shikimori"
+    var icon: UIImage? = UIImage(named: "shikimori")
```

Comment:
these variables should be at the top as the first three, since it's for conforming to the `Tracker` protocol

```suggestion
    let id = "shikimori"
    let name = "Shikimori"
    let icon = UIImage(named: "shikimori")
```